src/Libraries/PythonNodeModelsWpf/ScriptEditorWindow.xaml

```diff
@@ -312,6 +435,33 @@
                         </Button.Resources>
                     </Button>
                 </StackPanel>
+
+                <!-- The Warning Bar - activates on Esc if unsaved changes -->
+                <Grid Name="UnsavedChangesStatusBar"
+                        Grid.Row="4"
+                        Margin="0 -20 0 5"
+                        Height="95"
+                        Background="#EFD5C1"
+                        Visibility="Collapsed">
+                    <Grid.ColumnDefinitions>
+                        <ColumnDefinition Width="44"/>
+                        <ColumnDefinition Width="*"/>
+                        <ColumnDefinition Width="180"/>
+                    </Grid.ColumnDefinitions>
+                        <Border Background="#FAA21B" Grid.Column="0"> 
+                            <Image Source="/PythonNodeModelsWpf;component/Resources/alert.png" Width="24" Height="24"/>
+                        </Border>
+                        <StackPanel Orientation="Vertical" VerticalAlignment="Center" Grid.Column="1">
+                        <TextBlock Margin="15 5" FontSize="14"  FontFamily="{StaticResource ArtifaktElementBold}" Text="Are you sure you want to leave?"/>
+                        <TextBlock Margin="15 5" FontSize="14" FontFamily="{StaticResource ArtifaktElementRegular}" TextWrapping="WrapWithOverflow" Text="It looks like you are still working on this. You may lose any unsaved changes if you leave now"/>
```

Comment:
This text would need localization, can you add it to resources file.

Reply:
Resolved, thank you! It slipped under the radar.